core/src/main/java/org/elasticsearch/index/engine/RamAccountingSearcherFactory.java

```diff
@@ -52,22 +52,22 @@ public IndexSearcher newSearcher(IndexReader reader, IndexReader previousReader)
 
         // Construct a list of the previous segment readers, we only want to track memory used
         // by new readers, so these will be exempted from the circuit breaking accounting
-        final Set<SegmentReader> prevReaders;
+        final Set<IndexReader.CacheKey> prevReaders;
```

Comment:
++ maybe we leave a comment why we do it this way!?

Reply:
Yes very good idea